src/pre_defined_classes/mathematical_program/hard/q2/MainTest.java

```diff
@@ -0,0 +1,21 @@
+package pre_defined_classes.mathematical_program.hard.q2;
+
+import global.BaseTest;
+import global.variables.*;
+
+public class MainTest extends BaseTest {
+    //parsons
+    public Clause[] testSentence() {
```

Comment:
You might want to make this less hardcoded, just catch the values, and then check the math to make sure they return the correct value type and give better feedback, because right now if they use a distractor, they will just be told the code does not match the proper syntax/structure.

Reply:
Distractors:
 int xt = 1,000,000, x0 = 2; breaks
double t = Math.exp( xt / x0 ) / Math.exp( 1 + r ); Infinity
int t = Math.exp( xt / x0 ) / Math.exp( 1 + r ); breaks
int t = ( int )Math.exp( xt / x0 ) / Math.exp( 1 + r ); breaks
int t = ( double )( Math.exp( xt / x0 ) / Math.exp( 1 + r )); breaks
int t = Math.round( numerator / denominator ); breaks
double numerator = log( xt / x0 ); breaks
double denominator = log( 1 + r ); breaks

All but one of the distractors just breaks the program, there is no way for the math to be wrong except the infinity value, which still doesn't require a math test so I really don't see why I should make the math test

Reply:
I don't disagree with that point here, I just think that it is different then we have decided to write the tests, and that it is better to stick to the same sort of pattern